cmd/downloader/downloader/downloader_grpc_server.go

```diff
@@ -39,8 +39,11 @@ func (s *GrpcServer) Download(ctx context.Context, request *proto_downloader.Dow
 				return nil, err
 			}
 		}
+		var hash metainfo.Hash
```

Comment:
1. this looks dangerous, because later you add this "0x0000.." hash as magnet link. `magnet := mi.Magnet(&hash, nil)`
2. also you have 2 statements: `if it.TorrentHash == nil {` without return and then `if it.TorrentHash != nil { ` without return. Can't say what wrong with it, just smells wrong.

Reply:
maybe clearly separate this 2 scenarios: 
```
if it.TorrentHash != nil {
     ...
     continue
}
...
```

Reply:
doesn't this section create a torrent file and also assigns a torrent hash into `it.TorrentHash`: 
`if it.TorrentHash == nil { // seed new snapshot
			if err := BuildTorrentFileIfNeed(it.Path, s.d.SnapDir()); err != nil {
				return nil, err
			}
		}
`

Reply:
how it can assign to `it.TorrentHash` - `it` not passed as argument, only `it.Path` passed. Method just create torrent file on disk, that's it. 

This case is about new snapshots creation: Erigon creating new snapshot, it has no hash of snapshot - because hashing algorithm is area of Downloader responsibility. So, erigon sending request with file name and nil hash, Downloader does hash file - create .torrent file. Then adding this .torrent file to torrentClient - it will seed file by BitTorrent protocol.